parquet-hadoop/src/main/java/org/apache/parquet/hadoop/util/ContextUtil.java

```diff
@@ -251,17 +251,36 @@ public static Configuration getConfiguration(JobContext context) {
     }
   }
 
-  public static Counter getCounter(TaskInputOutputContext context,
-                                   String groupName, String counterName) {
-    return (Counter) invoke(GET_COUNTER_METHOD, context, groupName, counterName);
```

Comment:
I think this was intended to be a one-time startup cost. GET_COUNTER_METHOD was a static field that was the method. Now that this code finds them dynamically based on class it should be fine to remove GET_COUNTER_METHOD. Since this is done once per input split, I doubt the overhead is going to matter much.